api/base/filters.py

```diff
@@ -166,30 +290,39 @@ def get_queryset_from_request(self):
 
     def param_queryset(self, query_params, default_queryset):
         """filters default queryset based on query parameters"""
-        fields_dict = query_params_to_fields(query_params)
+        filters = self.parse_query_params(query_params)
         queryset = set(default_queryset)
-        if fields_dict:
-            for field_name, value in fields_dict.items():
-                if self.is_filterable_field(key=field_name):
-                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, value, default_queryset)))
-                else:
-                    raise InvalidFilterError
+        if filters:
+            for field_name, params in filters.iteritems():
+                for group in params:
+                    queryset = queryset.intersection(set(self.get_filtered_queryset(field_name, group, default_queryset)))
         return list(queryset)
 
-    def get_filtered_queryset(self, field_name, value, default_queryset):
+    def get_filtered_queryset(self, field_name, params, default_queryset):
         """filters default queryset based on the serializer field type"""
         field = self.serializer_class._declared_fields[field_name]
         field_name = field.source or field_name
 
         if isinstance(field, ser.SerializerMethodField):
-            return_val = [item for item in default_queryset if self.get_serializer_method(field_name)(item) == self.convert_value(value, field_name)]
+            return_val = [
+                item for item in default_queryset
+                if self.FILTERS[params['op']](self.get_serializer_method(field_name)(item), params['value'])
+            ]
         elif isinstance(field, ser.BooleanField):
```

Comment:
Is this branch necessary? It appears to be the same as what's in the `else` block